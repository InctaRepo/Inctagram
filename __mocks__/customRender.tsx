import React, { ReactElement } from 'react'

import { render, RenderOptions } from '@testing-library/react'
import { Provider } from 'react-redux'

import { store } from '@/store'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    // <ThemeProvider theme="light">
    //   <TranslationProvider messages={defaultStrings}>
    <Provider store={store}>{children}</Provider>
    //   </TranslationProvider>
    // </ThemeProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// eslint-disable-next-line import/export
export * from '@testing-library/react'
// eslint-disable-next-line import/export
export { customRender as render }
