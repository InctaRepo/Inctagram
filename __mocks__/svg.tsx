import React, { SVGProps, forwardRef } from 'react'

const SvgrMock = forwardRef<SVGSVGElement, SVGProps<SVGSVGElement>>((props, ref) => (
  <svg ref={ref} {...props} />
))

export const ReactComponent = SvgrMock
export default SvgrMock
