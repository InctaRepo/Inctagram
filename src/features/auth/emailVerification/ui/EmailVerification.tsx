import React from 'react'

import { useEmailRecoveryMutation } from '@/features/auth/emailVerification/service/emailRecovery'
import { Recovery } from '@/features/auth/recovery'

export const EmailVerification = () => {
  const [emailRecovery] = useEmailRecoveryMutation()

  return <Recovery recoveryMutation={emailRecovery} type={'email'} />
}
