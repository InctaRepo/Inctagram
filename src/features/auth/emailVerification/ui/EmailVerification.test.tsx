import React from 'react'

import { customRender as render } from '@/__mocks__/customRender'
import { act, fireEvent, screen, waitFor } from '@testing-library/react'

import { EmailVerification } from './EmailVerification'

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({ locale: 'en', query: { code: 'code' } }),
}))
const emailRecoveryMock = jest.fn()
const useEmailRecoveryMutation = jest.fn().mockReturnValue([emailRecoveryMock])

jest.mock('@/features/auth/emailVerification/service/emailRecovery', () => ({
  useEmailRecoveryMutation: () => useEmailRecoveryMutation(),
}))
jest.mock('@/ui/recaptcha', () => ({
  Recaptcha: jest.fn(({ onChange }) => {
    onChange(true)
  }),
}))
// ignore console.error
const consoleErrorSpy = jest.spyOn(console, 'error')

consoleErrorSpy.mockImplementation(() => {})
describe('EmailVerification', () => {
  it('submits the email recovery form', async () => {
    render(<EmailVerification />)

    await act(() =>
      fireEvent.change(screen.getByRole('email'), { target: { value: 'test@example.com' } })
    )
    await act(() => fireEvent.click(screen.getByRole('button', { name: 'Send Link' })))
    await waitFor(() => {
      expect(emailRecoveryMock).toHaveBeenCalledWith({ email: 'test@example.com' })
    })
  })
  it('snapshot EmailVerification', () => {
    const snapshot = render(<EmailVerification />)

    waitFor(() => expect(snapshot).toMatchSnapshot())
  })
})
