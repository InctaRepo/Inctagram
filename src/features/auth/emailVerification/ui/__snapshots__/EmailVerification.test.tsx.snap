// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailVerification snapshot EmailVerification 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style=""
  >
    <div>
      <div
        class="primary"
      >
        <div
          class="card main"
        >
          <p
            class="h1 primary  title"
          >
            Resend verification link
          </p>
          <form
            class="form"
          >
            <div
              class="email "
            >
              <div
                class="textFieldWrap"
              >
                <label
                  class="label undefined"
                  data-testid="email"
                  for="email"
                >
                  Email
                </label>
                <input
                  class="textField undefined undefined"
                  id="email"
                  name="email"
                  placeholder="Epam@epam.com"
                  role="email"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <p
              class="regular14 primary  hint"
            >
              Enter your email address and we will send you further instructions
            </p>
            <button
              class="primary  submit"
              type="submit"
            >
              <p
                class="h3 primary  undefined"
              >
                Send Link
              </p>
            </button>
            <p
              class="regular14 primary  answer"
            >
              The link has been sent by email. If you dont receive an email send link again
            </p>
            <button
              class="primary  repeat"
              type="submit"
            >
              <p
                class="regular16 primary  undefined"
              >
                Send Link Again
              </p>
            </button>
            <button
              class="link  back"
              color="$color-accent-500"
              type="button"
            >
              <p
                class="bold16 primary  undefined"
              >
                Back to Sign In
              </p>
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="primary"
    >
      <div
        class="card main"
      >
        <p
          class="h1 primary  title"
        >
          Resend verification link
        </p>
        <form
          class="form"
        >
          <div
            class="email "
          >
            <div
              class="textFieldWrap"
            >
              <label
                class="label undefined"
                data-testid="email"
                for="email"
              >
                Email
              </label>
              <input
                class="textField undefined undefined"
                id="email"
                name="email"
                placeholder="Epam@epam.com"
                role="email"
                type="text"
                value=""
              />
            </div>
          </div>
          <p
            class="regular14 primary  hint"
          >
            Enter your email address and we will send you further instructions
          </p>
          <button
            class="primary  submit"
            type="submit"
          >
            <p
              class="h3 primary  undefined"
            >
              Send Link
            </p>
          </button>
          <p
            class="regular14 primary  answer"
          >
            The link has been sent by email. If you dont receive an email send link again
          </p>
          <button
            class="primary  repeat"
            type="submit"
          >
            <p
              class="regular16 primary  undefined"
            >
              Send Link Again
            </p>
          </button>
          <button
            class="link  back"
            color="$color-accent-500"
            type="button"
          >
            <p
              class="bold16 primary  undefined"
            >
              Back to Sign In
            </p>
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
  "user": {
    "clear": [Function],
    "click": [Function],
    "copy": [Function],
    "cut": [Function],
    "dblClick": [Function],
    "deselectOptions": [Function],
    "hover": [Function],
    "keyboard": [Function],
    "paste": [Function],
    "pointer": [Function],
    "selectOptions": [Function],
    "setup": [Function],
    "tab": [Function],
    "tripleClick": [Function],
    "type": [Function],
    "unhover": [Function],
    "upload": [Function],
  },
}
`;
